src/hooks/useHandleModalClick.js

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from "react";
+
+/**
+ *  @hooks
+ *  Modal 외부클릭시 handleModalClose 실행
+ */
+
+export const useHandleModalClick = (modalRef, handleModalClose) => {
+  useEffect(() => {
+    // OutSide Click => Modal Close
+    const handleClickOutside = (e) => {
+      if (modalRef.current && !modalRef.current.contains(e.target)) {
+        handleModalClose();
+      }
+    };
+
+    // ESC keydown => Modal Close
+    const handleKeyDownEsc = (e) => {
+      if (e.key === "Escape") {
+        handleModalClose();
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+    document.addEventListener("keydown", handleKeyDownEsc);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+      document.removeEventListener("keydown", handleKeyDownEsc);
+    };
+  }, [modalRef, handleModalClose]);
```

Comment:
P2:
함수들은 useEffect callback 부분이 아니라 외부에 있는게 더 가독성이 좋을 것 같아요.

```suggestion
// OutSide Click => Modal Close
  const handleClickOutside = useCallback(
    (e) => {
      if (modalRef.current && !modalRef.current.contains(e.target)) {
        handleModalClose();
      }
    },
    [handleModalClose, modalRef]
  );

  // ESC keydown => Modal Close
  const handleKeyDownEsc = useCallback(
    (e) => {
      if (e.key === "Escape") {
        handleModalClose();
      }
    },
    [handleModalClose]
  );

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    document.addEventListener("keydown", handleKeyDownEsc);

    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
      document.removeEventListener("keydown", handleKeyDownEsc);
    };
  }, [handleClickOutside, handleKeyDownEsc]);
```